### PR TITLE
V0.5 wip

### LINKED ISSUE #207: Consistency Issues within symmetric ciphers
Need to make AES-CFB1 work like TDES-CFB1 since the pt/ct format is completely different.
Need to make all symmetric ciphers use dataLength in capabilities to be consistent.

### LINKED ISSUE #244: Update `failureTest`/`decryptFail` to be "in the positive"
I'm not sure of the correct way to phrase this, but it's likely to be less confusing to change instances such as:

```
"decryptFail" : true // test is a failure
"decryptFail" : false // test is a pass
"failureTest" : true // test is a failure
"failureTest" : false // test is a pass
"result" : "pass"
"result" : "passed"
"result" : "fail"
"result" : "failed"
```

to

```
"testPassed" : true // test is a pass
"testPassed" : false // test is a failure
```

The `isSuccessFulTest` negates potential double negatives, as well as changes the language to be more general, not specific to "decrypt/encrypt".

Thoughts?  I'm not super set on `isSuccessfulTest`, but I feel like the current should be updated.

### LINKED ISSUE #254: Investigate removal of IV from CTR mode tests
Quote from another issue from Stephan...

> If the CAVS server would perform the steps outlined above, all would benefit as follows:
>1. The developer would not need a debug build of their code.
>2. The developer could simply perform a standard en/decryption (i.e. AFT) just like for the other modes.
>3. CAVS would get a result from the implementation that is shipped to customers (i.e. no debug build, no simulation testing, etc.)

### LINKED ISSUE #291: DRBG valid values
I know that some of the acceptable values for DRBG are outlined in other NIST docs... is it intended to be more explicitly outlined in the ACVP specs somewhere down the line? If so, the RSA spec could use some more detail in this area as well :) not a blocking issue, just thought I'd ask since it could simplify the process of constructing a client registration if all the details were in one document.

Thank you!

### LINKED ISSUE #344: CFB1 spec request: use different keyword for pt / ct
The JSON keywords imply a data type. For the keywords of ct/pt, the data type is the hex representation of bytes.

For CFB-1, however, pt and ct are supposed to be representations of one bit. To support generic parsers, may I suggest to use a different key word that can be associated with the different data type?

Thanks

### PR DESCRIPTION
**Pull request by Kritner**
Initial pass on updates to prepare for the 0.5 version of the spec.

Summary of changes:
* new property of tgId
* tgId added to prompt and results file (restructures results file to be hierarchical to support IUT sending group level properties)
* "result", "decryptFail", and other similarly named properties renamed to a consistent `testPassed` boolean property, fixes #244
* DRBG additional table based on allowed values, fixes #291 
* AES-CFB1 changed to use the consistent BItString (hex string) rather than individual bits. fixes #344, fixes #207
* Began #122 - WIP
* Removes IV from Counter tests, fixes #254

### CODE REVIEW

**Comment by smuellerDD** [comment]
@Kritner @hbooth What is the status of the v0.5 REST API interface? Is it already live so that I can start development against it?

**Comment by hbooth** [comment]
@smuellerDD We are targeting early October to have the 0.5 API fully available. I would expect that parts of the API will become available in late September. We have been and are continuing to work on some of the back support so that we will be able to successfully manage the environment.

File: artifacts/draft-celi-block-ciph-00.txt [hunk 1]
```diff
@@ -1,3528 +1,4200 @@
...
-       "keyLen" : 128,
-       "tests" : [ {
-         "tcId" : 1,
-         "pt" : "",
-         "key" : "BCDD633A9FD07CC2EAED1C13151F59CF",
-         "aad" : "4EB22A3D97AA149AF30C6D1E4E63BFF7"
-       }, {
-         "tcId" : 2,
-         "pt" : "",
-         "key" : "97D529AFB0B1B6276B144
```
Comment (bfussell) [inline artifacts/draft-celi-block-ciph-00.txt]: Is there additional work to address issue #254/#245 or does this cover it ?

File: src/draft-celi-block-ciph-00.xml [hunk 2]
```diff
@@ -27,8 +27,8 @@
 <?rfc subcompact="no" ?>
 <!-- keep one blank line between list items -->
 <!-- end of list of popular I-D processing instructions -->
-<rfc category="info" docName="draft-celi-block-ciph-00" ipr="trust200902">
-    <!-- category values: std, bcp, info, exp, and historic
+<rfc category="info" docName="draft-ietf-acvp-subsym-0.5" ipr="trust200902">
```
[the remark this answers is not in the document]
Comment (atvassilev) [reply src/draft-celi-block-ciph-00.xml]: In my opinion, the new file name "draft-ietf-acvp-subsym-0.5" does not meet the IETF requirements. The old one, draft-celi-block-ciph-00, does and should be kept.

File: artifacts/draft-celi-block-ciph-00.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (atvassilev) [reply artifacts/draft-celi-block-ciph-00.txt]: Sure, I suggest identifying a few people (2-3) as authors, the others can be listed in the acknowledgements section. Ideally, there should be one editor but in some cases two work too. We should avoid having more than two people as editors on a single spec.

Comment (celic) [reply artifacts/draft-celi-block-ciph-00.txt]: #254 is accounted for here. There may not be examples of it (or the examples may be out of date). I'll double check this and make the needed changes.

File: src/draft-celi-block-ciph-00.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (celic) [reply src/draft-celi-block-ciph-00.xml]: Ok. I wasn't sure on updating the name like other specs in order to account for the version. I'll revert this.

**Comment by celic** [comment]
I made sure all the symmetric examples have testTypes in the group prompt. As well I added examples for AES-XPN which were seemingly missing. I removed the IV from the Counter tests as those are extraneous now. I cleaned up some extra line breaks off of tables as well. Thank you for the comments.

**Comment by celic** [comment]
I have updated the examples to use `<acvp-version>` instead of `"0.5"`. It is up to the reader to determine what version of the protocol they are using.

**Pull request merged**